tasks/build.py

```diff
@@ -1,30 +1,113 @@
+    log("scripts_dir '%s'" % (scripts_dir))
+    local_tmp = config.get_section('buildenv').get('local_tmp')
+    log("local_tmp '%s'" % (local_tmp))
+    build_job_script = config.get_section('buildenv').get('build_job_script')
+    log("build_job_script '%s'" % (build_job_script))
+    submit_command = config.get_section('buildenv').get('submit_command')
+    log("submit_command '%s'" % (submit_command))
 
-def build_easystack_from_pr(pr, request):
-    event_id = request.headers['X-Github-Delivery']
-    jobdir = os.path.join(JOBS_DIR, event_id)
-    os.makedirs(jobdir)
-    #shutil.copytree(SCRIPTS_DIR, jobdir)
-    # TODO: checkout the branch that belongs to the PR
+    # [softwaretargets]
+    # software_file currently not being used (may be later to limit what files will be processed)
+    software_file = config.get_section('softwaretargets').get('software_file')
+    log("software file '%s'" % (software_file))
+    whitelist_versions = json.loads(config.get_section('softwaretargets').get('whitelist_versions'))
+    log("whitelisted versions '%s'" % (json.dumps(whitelist_versions)))
+
+    # [architecturetargets]
+    arch_target_map = json.loads(config.get_section('architecturetargets').get('arch_target_map'))
+    log("arch target map '%s'" % (json.dumps(arch_target_map)))
+
+    # get event id and create main directory for event and a temporary dir
+    # inside that (the latter uses mktemp to create a new job every time the
+    # event is being redelivered, i.e., useful while developing)
+    event_id = event_info['id']
+    jobs_dir = os.path.join(jobs_base_dir, event_id)
+    if not os.path.exists(jobs_dir):
+        os.makedirs(jobs_dir)
+    job_run_dir = tempfile.mkdtemp(dir=jobs_dir)
+    log("created temporary job run directory '%s'" % (job_run_dir))
+
+    # TODO: checkout the branch that belongs to the PR --> what's actually missing???
     # PyGitHub doesn't seem capable of doing that (easily);
     # for now, keep it simple and just download the easystack file
     gh = github.get_instance()
     repo = gh.get_repo(pr.head.repo.full_name)
-    easystack = repo.get_contents(EASYSTACK_PATH, ref=pr.head.ref)
-    with open(os.path.join(jobdir, os.path.basename(EASYSTACK_PATH)), 'w') as easystack_file:
-        easystack_file.write(easystack.decoded_content.decode('UTF-8'))
+    log("pr.head.repo.full_name '%s'" % (pr.head.repo.full_name))
+    log("pr.head.ref '%s'" % (pr.head.ref))
+
+    # - determine files to proceed and prepare job dirs (essentially we
+    #   may have gotten multiple easystack files (N) via a single PR and
+    #   the bot may be configured to build for different
+    #   (micro)architectures (M); resulting in N x M jobs)
+    # - the N files may be filtered by white listed versions (bot config)
+    # - we assume that the full path to the easystack file includes
+    #   an EESSI version, e.g., 2021.12/softwarelist.yaml
+    jobs = []
+    for pr_file in repo.get_pull(pr.number).get_files():
```

Comment:
We don't want to download individual files eventually, we just want to get a checkout of the entire PR branch, and use that as a working directory for the job.

With the current approach of just "reacting" to specific easystack files, this is fine, but it can probably be cleaned up quite a bit...

Reply:
Created issue #5 to take care of this.